substrate/frame/election-provider-multi-block/src/lib.rs

```diff
@@ -607,15 +615,17 @@ pub mod pallet {
 		/// The origin of this call must be [`Config::AdminOrigin`].
 		///
 		/// See [`AdminOperation`] for various operations that are possible.
-		#[pallet::weight(T::WeightInfo::manage())]
+		#[pallet::weight(T::WeightInfo::manage_set()
+			.max(T::WeightInfo::manage_fallback())
+			.max(T::WeightInfo::export_terminal())
+		)]
 		#[pallet::call_index(0)]
 		pub fn manage(origin: OriginFor<T>, op: AdminOperation<T>) -> DispatchResultWithPostInfo {
-			use crate::verifier::Verifier;
-			use sp_npos_elections::EvaluateSupport;
-
-			let _ = T::AdminOrigin::ensure_origin(origin);
 			match op {
 				AdminOperation::EmergencyFallback => {
+					// Fallback, if present, is a "trusted" way to elect a validator set, and
+					// therefore can be dispatched by `ManagerOrigin`.
+					T::ManagerOrigin::ensure_origin(origin)?;
```

Comment:
Feel a little bit paranoid about keeping the origin check inside match arms. Might be cleaner to keep separate extrinsics for admin and manager?